crates/bevy_ecs/src/archetype.rs

```diff
@@ -328,11 +329,12 @@ pub struct Archetype {
     table_id: TableId,
     edges: Edges,
     entities: Vec<ArchetypeEntity>,
-    components: ImmutableSparseSet<ComponentId, ArchetypeComponentInfo>,
+    components: HashMap<ComponentId, ArchetypeComponentInfo>,
```

Comment:
Should we also add a `Vec<(ComponentId, ArchetypeComponentInfo)>` to maintain a sorted list of component ids?

Reply:
Just having a `Vec<ComponentId>` is an option as well.

Reply:
If you want a sorted list, consider `BTreeSet`. It automatically sorts its entries and has $O(log(n))$ for most operations.